cli-toolchain/graphql.md

```diff
@@ -1874,215 +2166,740 @@ type Comment {
+* `matchPhrase` - searches using the Elasticsearch's [Match Phrase Query](https://www.elastic.co/guide/en/elasticsearch/reference/6.2/query-dsl-match-query-phrase.html) to filter the documents in the search query.
+* `matchPhrasePrefix` - This uses the Elasticsearch's [Match Phrase Prefix Query](https://www.elastic.co/guide/en/elasticsearch/reference/6.2/query-dsl-match-query-phrase-prefix.html) to filter the documents in the search query.
+* `multiMatch` - Corresponds to the Elasticsearch [Multi Match Query](https://www.elastic.co/guide/en/elasticsearch/reference/6.2/query-dsl-multi-match-query.html).
+* `exists` - Corresponds to the Elasticsearch [Exists Query](https://www.elastic.co/guide/en/elasticsearch/reference/6.2/query-dsl-exists-query.html).
+* `wildcard` - Corresponds to the Elasticsearch [Wildcard Query](https://www.elastic.co/guide/en/elasticsearch/reference/6.2/query-dsl-wildcard-query.html).
+* `regexp` - Corresponds to the Elasticsearch [Regexp Query](https://www.elastic.co/guide/en/elasticsearch/reference/6.2/query-dsl-regexp-query.html).
+
+
+For example, you can filter using the wildcard expression to search for posts using the following `wildcard` query:
+
+```graphql
+query SearchPosts {
+  searchPost(filter: { title: { wildcard: "S*Elasticsearch!" }}) {
+    items {
+      id
+      title
+    }
+  }
+}
+```
+
+The above query returns all documents whose `title` begins with `S` and ends with `Elasticsearch!`.
+
+Moreover you can use the `filter` parameter to pass a nested `and`/`or`/`not` condition. By default, every operation in the filter properties is *AND* ed. You can use the `or` or `not` properties in the `filter` parameter of the search query to override this behavior. Each of these operators (`and`, `or`, `not` properties in the filter object) accepts an array of searchable types which are in turn joined by the corresponding operator. For example, consider the following search query:
+
+```graphql
+query SearchPosts {
+  searchPost(filter: {
+    title: { wildcard: "S*" }
+    or: [
+      { createdAt: { eq: "08/20/2018" } },
+      { updatedAt: { eq: "08/20/2018" } }
+    ]
+  }) {
+    items {
+      id
+      title
+    }
+  }
+}
+```
+
+Assuming you used the `createPost` mutation to create new posts with `title`, `createdAt` and `updatedAt` values, the above search query will return you a list of all `Posts`, whose `title` starts with `S` _and_ have `createdAt` _or_ `updatedAt` value as `08/20/2018`.
+
+Here is a complete list of searchable operations per GraphQL type supported as of today:
+
+| GraphQL Type        | Searchable Operation           |
+|-------------:|:-------------|
+| String      | `ne`, `eq`, `match`, `matchPhrase`, `matchPhrasePrefix`, `multiMatch`, `exists`, `wildcard`, `regexp` |
+| Int     | `ne`, `gt`, `lt`, `gte`, `lte`, `eq`, `range`      |
+| Float | `ne`, `gt`, `lt`, `gte`, `lte`, `eq`, `range`      |
+| Boolean | `eq`, `ne`      |
+
+
+## 17 Data Access Patterns
```

Comment:
Are these supposed to go into a separate md file per the link above?

Reply:
Not unless you think they should go into a separate file, for now they link to this part of the page.